lib/Sema/CSSimplify.cpp

```diff
@@ -2063,13 +2063,27 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
     if (path.back().is<LocatorPathElt::PatternMatch>()) {
       return true;
     } else if (path.size() > 1) {
-      // sub-pattern matching as part of the enum element matching
-      // where sub-element is a tuple pattern e.g.
-      // `case .foo((a: 42, _)) = question`
-      auto lastIdx = path.size() - 1;
-      if (path[lastIdx - 1].is<LocatorPathElt::PatternMatch>() &&
-          path[lastIdx].is<LocatorPathElt::FunctionArgument>())
-        return true;
+        int patternMatchIdx = 0;
+        for (int i = 0; i < (int)path.size(); i++) {
+            if (path[i].is<LocatorPathElt::PatternMatch>()) {
+                patternMatchIdx = i;
+            }
+        }
```

Comment:
We could simplify that to 
```suggestion
              auto it = llvm::find_if(path, [](LocatorPathElt &elt) {
        return elt.is<LocatorPathElt::PatternMatch>();
      });
      if (it == path.end())
        return false; 
```
Also, was checking `LocatorPathElt::FunctionArgument` and `LocatorPathElt::TupleElement` really needed? Or  only returning `LocatorPathElt::PatternMatch` found or not break something on test suit?

Reply:
I didn't fully understand what you just said,
I never thought it would be this easy.
Fixed it and the test case seems to pass.

Thanks for the code suggestions as well.

Reply:
Awesome, if all tests pass looks like we don't need the more specific check.
We may only need to remove a couple of redundant logic in there =]